share/readonlyadmin.py

```diff
@@ -0,0 +1,56 @@
+## adopted from http://mike.hostetlerhome.com/blog/2012/11/13/add-a-read-only-role-to-django-admin/
+from django.contrib import admin
+from django.core.exceptions import PermissionDenied
+
+
+class ReadOnlyAdmin(admin.ModelAdmin):
+    """
+    """
+
+    def has_add_permission(self, request, obj=None):
+        """
+        Arguments:
+        - `request`:
+        - `obj`:
+        """
+        return self.__user_is_readonly(request)
+
+    def has_delete_permission(self, request, obj=None):
+        """
+        Arguments:
+        - `request`:
+        - `obj`:
+        """
+        return self.__user_is_readonly(request)
+
+    def get_actions(self, request):
+
+        actions = super(ReadOnlyAdmin, self).get_actions(request)
+
+        if self.__user_is_readonly(request):
+            if 'delete_selected' in actions:
+                del actions['delete_selected']
+
+        return actions
+
+    def change_view(self, request, object_id, extra_context=None):
+
+        if self.__user_is_readonly(request):
```

Comment:
@chrisseto, I'm not following what you mean here.  Are you saying that `self.readonly_fields = self.user_readonly` should be something like `self.readonly_fields = self.model._meta.get_all_field_names()`*

*not actually correct